h2o-core/src/test/java/water/fvec/TestFrameBuilderTest.java

```diff
@@ -0,0 +1,122 @@
+package water.fvec;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.TestUtil;
+
+public class TestFrameBuilderTest extends TestUtil {
+  @BeforeClass
+  public static void setup() { stall_till_cloudsize(1); }
+
+  @Test
+  public void testEmpty(){
+    Frame fr = new TestFrameBuilder().build();
+
+    assert fr.vecs().length == 0;
+    assert fr.numCols() == 0;
+    assert fr.numRows() == 0;
+    fr.remove();
+  }
+
+  @Test
+  public void testName(){
+    Frame fr = new TestFrameBuilder()
+            .withName("FrameName")
+            .build();
+
+    assert fr._key.toString().equals("FrameName");
```

Comment:
It is empty frame. I haven't done the check here because it's already in the test above this one. I wanted to make tests small with checking only one feature at a time. Can add the check also here though